### PR TITLE
Player time readout changes

### PR DESCRIPTION
**Pull request by Kyli3Boi**
This is various changes to the way that player times are displayed. All examples of changes can be found in #1 

Im no coder but iv done my best. Please review and i hope you all approve.

### CODE REVIEW

**Review by dPexxIAM — APPROVED** [review]

File: addons/sourcemod/scripting/surftimer/misc.sp [hunk 1]
```diff
@@ -4223,108 +4311,206 @@ stock void StyleFinishedMsgs(int client, int style)
...
+		// Map style SR, time difference formatting
+		RecordDiff = g_fRecordStyleMapTime[style] - g_fFinalTime[client];
+		FormatTimeFloat(client, RecordDiff, 3, szRecordDiff, 32);
+		if (RecordDiff > 0.0)
+		{
+		Format(szRecordDiff, 32, "-%s", szRecordDiff);
+		}
+		else
+		{
+		Format(szRecordDiff, 32, "+%s", szRecordDiff);
```
Comment (Bara) [inline addons/sourcemod/scripting/surftimer/misc.sp:4328]: Missing tab

File: addons/sourcemod/scripting/surftimer/misc.sp [hunk 2]
```diff
@@ -4223,108 +4311,206 @@ stock void StyleFinishedMsgs(int client, int style)
...
+	RecordDiff = g_fStyleBonusFastest[style][zGroup] - g_fFinalTime[client];
+	FormatTimeFloat(client, RecordDiff, 3, szRecordDiff, 54);
+	if (RecordDiff > 0.0)
 	{
-		PlayRecordSound(2);
+	Format(szRecordDiff, 54, "-%s", szRecordDiff);
+	}
+	else
+	{
+	Format(szRecordDiff, 54, "+%s", szRecordDiff);
```
Comment (Bara) [inline addons/sourcemod/scripting/surftimer/misc.sp:4457]: Missing tab

File: addons/sourcemod/scripting/surftimer/misc.sp [hunk 3]
```diff
@@ -4223,108 +4311,206 @@ stock void StyleFinishedMsgs(int client, int style)
...
 
-		RecordDiff = g_fStyleOldBonusRecordTime[style][zGroup] - g_fFinalTime[client];
-		FormatTimeFloat(client, RecordDiff, 3, szRecordDiff, 54);
-		Format(szRecordDiff, 54, "-%s", szRecordDiff);
+	// Player beat bonus style SR, time difference formatting
+	RecordDiff2 = g_fStyleOldBonusRecordTime[style][zGroup] - g_fFinalTime[client];
+	FormatTimeFloat(client, RecordDiff2, 3, szRecordDiff2, 54);
+	if (RecordDiff2 > 0.0)
+	{
+	Format(szRecordDiff2, 54, "-%s", szRecordDiff2);
```
Comment (Bara) [inline addons/sourcemod/scripting/surftimer/misc.sp:4465]: Missing tab

File: addons/sourcemod/scripting/surftimer/misc.sp [hunk 4]
```diff
@@ -4223,108 +4311,206 @@ stock void StyleFinishedMsgs(int client, int style)
...
 	}
-	if (g_bBonusFirstRecord[client] && g_bBonusSRVRecord[client])
+	else
 	{
-		CPrintToChatAll("%t", "Misc37", g_szChatPrefix, szName, g_szZoneGroupName[zGroup], g_szStyleRecordPrint[style]);
-		if (g_StyletmpBonusCount[0][zGroup] == 0)
-			CPrintToChatAll("%t", "Misc38", g_szChatPrefix, szName, g_szZoneGroupName[zGroup], g_szStyleRecordPrint[style], g_szFinalTime[client], g_szFinalTime[client]);
-		else
-			CPrintToChatAll("%t", "Misc39", g_szChatPrefix, szName, g_szZoneGroupName[zGroup], g_szStyleRecordPrint[style], g_szFinalTime[client], szRecordDiff, g_StyleMapRankBonus[style][zGroup][client], g_iStyleBonusCount[style][zGroup], g_szFinalTime[client]);
+	Format(szRecordDiff2, 54, "+%s", szRecordDiff2);
```
Comment (Bara) [inline addons/sourcemod/scripting/surftimer/misc.sp:4469]: Missing tab

**Review by Bara — CHANGES_REQUESTED** [review]

File: addons/sourcemod/scripting/surftimer/buttonpress.sp [hunk 5]
```diff
@@ -553,18 +556,34 @@ public void CL_OnEndTimerPress(int client)
...
+				float RecordDiff;
+				char szRecordDiff[32];
 				int count = g_StyleMapTimesCount[style];
 
+				// Map style SR, time difference formatting
+				RecordDiff = g_fRecordStyleMapTime[style] - g_fFinalTime[client];
+				FormatTimeFloat(client, RecordDiff, 3, szRecordDiff, 32);
+				if (RecordDiff > 0.0)
+				{
+				Format(szRecordDiff, 32, "-%s", szRecordDiff);
```
[the remark this answers is not in the document]
Comment (Kyli3Boi) [reply addons/sourcemod/scripting/surftimer/buttonpress.sp:570]: Fixed

File: addons/sourcemod/scripting/SurfTimer.sp [hunk 6]
```diff
@@ -460,6 +460,7 @@ int g_OldStyleMapRank[MAX_STYLES][MAXPLAYERS + 1];
 float g_fPersonalStyleRecord[MAX_STYLES][MAXPLAYERS + 1];
 char g_szPersonalStyleRecord[MAX_STYLES][MAXPLAYERS + 1][256];
 float g_fRecordStyleMapTime[MAX_STYLES];
+float g_fOldRecordStyleMapTime[MAX_STYLES];
```
[the remark this answers is not in the document]
Comment (Kyli3Boi) [reply addons/sourcemod/scripting/SurfTimer.sp:463]: This is something that i would need assistance with as i have had a look at it and im not very confident that i understand what needs to be changed.

[the remark this answers is not in the document]
Comment (Bara) [reply addons/sourcemod/scripting/SurfTimer.sp:463]: https://github.com/surftimer/Surftimer-Official/pull/258/commits/adf2267d7396fea23debf14f42553b8e0efab85e

[the remark this answers is not in the document]
Comment (Bara) [reply addons/sourcemod/scripting/SurfTimer.sp:463]: It's just replace of `g_fOldRecordMapTime` to `g_fOldRecordStyleMapTime[0]`, 0 is for style "Normal".

**Review by Bara — APPROVED** [review]
I'm fine with it, if it works as it should.

**Comment by Kyli3Boi** [comment]
Oh thanks Bara, I didn't think that was all I had to do... was thinking I would have to add/change a lot more stuff

**Pull request merged**